src/clone.rs

```diff
@@ -412,7 +407,7 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn should_rewrite_checkpoint_if_uninitialized_checkpoint_is_invalid() {
+    async fn should_fail_retry_if_uninitialized_checkpoint_is_invalid() {
```

Comment:
@rodesai please pay closer attention to this test, since it slightly changes behavior

This is basically case where we create "ephemeral source checkpoint" and it expires before retry. If we'd create yet another source checkpoint, we start running into corner cases (different set of external dbs, potentially leaked final checkpoints, ...).

In case ephemeral checkpoint expires we should just fail, otherwise we break idempotency. I think it would be worth to make "timeouts" for ephemeral checkpoints configurable.